datadog/build.go

```diff
@@ -52,10 +52,12 @@ func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 		nativeImage := cr.ResolveBool("BP_NATIVE_IMAGE")
 		result.Layers = append(result.Layers, NewJavaAgent(agentDependency, dc, b.Logger, nativeImage))
 
-		h, be := libpak.NewHelperLayer(context.Buildpack, "toggle")
-		h.Logger = b.Logger
-		result.Layers = append(result.Layers, h)
-		result.BOM.Entries = append(result.BOM.Entries, be)
+		if (!nativeImage) {
+			h, be := libpak.NewHelperLayer(context.Buildpack, "toggle")
+			h.Logger = b.Logger
+			result.Layers = append(result.Layers, h)
+			result.BOM.Entries = append(result.BOM.Entries, be)
+		}
```

Comment:
Can you add a test case to cover this? You're adding a new conditional here, so it's probably not being tested with the existing test suite.